### PR TITLE
clean up #[allow] and generated rustfmt::skip

### PR DESCRIPTION
**Pull request by youknowone**

### CODE REVIEW

File: lalrpop/src/lr1/codegen/base.rs [hunk 1]
```diff
@@ -158,16 +158,13 @@ impl<'codegen, 'grammar, W: Write, C> CodeGenerator<'codegen, 'grammar, W, C> {
...
-        rust!(self.out, "#[cfg_attr(rustfmt, rustfmt_skip)]");
-        rust!(self.out, "mod {}parse{} {{", self.prefix, self.start_symbol);
-
-        // these stylistic lints are annoying for the generated code,
-        // which doesn't follow conventions:
+        rust!(self.out, "#[rustfmt::skip]");
         rust!(
             self.out,
-            "#![allow(non_snake_case, non_camel_case_types, unused_mut, unused_variables, \
+            "#[allow(non_snake_case, non_camel_case_types, unused_mut, unused_variables, \
```
Comment (youknowone) [inline lalrpop/src/lr1/codegen/base.rs:164]: Not much reason to move this up, but keeping `#[allow]` and `#[rustfmt::skip]` at the same level will make sense

File: lalrpop-test/src/lib.rs [hunk 2]
```diff
@@ -20,7 +20,6 @@ macro_rules! lalrpop_mod_test {
     ($(#[$attr:meta])* $vis:vis $modname:ident) => {
         lalrpop_mod!(
             #[allow(clippy::ptr_arg)]
```
Comment (Pat-Lafon) [inline lalrpop-test/src/lib.rs:22]: In doc/parcal/lalrpop/src/main.rs, both clippy lints(vec_box and ptr_arg) were removed. Can the same be done here?

File: lalrpop/src/lr1/codegen/base.rs (hunk 1 above)
Comment (Pat-Lafon) [reply lalrpop/src/lr1/codegen/base.rs:164]: Somewhere else I had suggested using `#[automatically_derived]`, would that work here?

Comment (youknowone) [reply lalrpop/src/lr1/codegen/base.rs:164]: It only works for `impl`. This is `mod`, so it doesn't work.

Comment (Pat-Lafon) [reply lalrpop/src/lr1/codegen/base.rs:164]: Shucks!

**Pull request merged**